hamilton/plugins/huggingface_extensions.py

```diff
@@ -0,0 +1,206 @@
+    task = "deprecated"
+    streaming: bool = False
+    num_proc: Optional[int] = None
+    storage_options: Optional[Dict] = None
+    config_kwargs: Optional[Dict] = None
+
+    @classmethod
+    def applicable_types(cls) -> Collection[Type]:
+        return list(HF_types)
+
+    def _get_loading_kwargs(self) -> dict:
+        # Puts kwargs in a dict
+        kwargs = dataclasses.asdict(self)
+        # we send path separately
+        del kwargs["path"]
+        config_kwargs: Optional[dict] = kwargs.pop("config_kwargs", None)
+        if config_kwargs:
+            # add config kwargs as needed.
+            kwargs.update(config_kwargs)
+
+        # need to pass in name
+        kwargs["name"] = kwargs.pop("dataset_name", None)
+
+        return kwargs
+
+    def load_data(self, type_: Type) -> Tuple[Union[HF_types], dict[str, Any]]:
+        """"""
+        ds = load_dataset(self.path, **self._get_loading_kwargs())
+        is_dataset = isinstance(ds, Dataset)
+        f_meta = {"path": self.path}
+        ds_meta = {"rows": ds.num_rows, "columns": ds.column_names}
+        if is_dataset:
+            ds_meta["size_in_bytes"] = ds.size_in_bytes
+            ds_meta["features"] = ds.features.to_dict()
+        return ds, {"file_metadata": f_meta, "dataset_metadata": ds_meta}
+
+    @classmethod
+    def name(cls) -> str:
+        return "hf_dataset"
+
+
+@dataclasses.dataclass
+class HuggingFaceDSParquetSaver(DataSaver):
+    path_or_buf: Union[PathLike, BinaryIO]
+    batch_size: Optional[int] = None
+    parquet_writer_kwargs: Optional[dict] = None
+
+    @classmethod
+    def applicable_types(cls) -> Collection[Type]:
+        return list(HF_types)
+
+    @classmethod
+    def applies_to(cls, type_: Type[Type]) -> bool:
+        return type_ in HF_types
+
+    def _get_saving_kwargs(self) -> dict:
+        # Puts kwargs in a dict
+        kwargs = dataclasses.asdict(self)
+        # but we send it separately
+        del kwargs["path_or_buf"]
+        parquet_writer_kwargs: Optional[dict] = kwargs.pop("parquet_writer_kwargs", None)
+        if parquet_writer_kwargs:
+            # add config kwargs as needed.
+            kwargs.update(parquet_writer_kwargs)
+
+        return kwargs
+
+    def save_data(self, ds: Union[HF_types]) -> Dict[str, Any]:
+        is_dataset = isinstance(ds, Dataset)
+        ds.to_parquet(self.path_or_buf, **self._get_saving_kwargs())
+        ds_meta = {
+            "rows": ds.num_rows,
+            "columns": ds.column_names,
+        }
+        if is_dataset:
+            ds_meta.update({"size_in_bytes": ds.size_in_bytes, "features": ds.features.to_dict()})
+        if isinstance(self.path_or_buf, BinaryIO):
+            f_meta = {}
+        else:
+            f_meta = (utils.get_file_metadata(self.path_or_buf),)
+        return {"file_metadata": f_meta, "dataset_metadata": ds_meta}
+
+    @classmethod
+    def name(cls) -> str:
+        return "parquet"
+
+
+if lancedb is not None:
```

Comment:
We should come up with a better way to do this... I guess this is fine for now, but it seems like we include this in lanceDB and only have the `applicable_types` include Dataset if we have Huggingface installed...

Reply:
that's the same though. So I thought this was more HF specific so should live here.